AirCasting/SearchAndFollow/CompleteScreen/CompleteScreenViewModel.swift

```diff
@@ -0,0 +1,32 @@
+// Created by Lunar on 22/02/2022.
+//
+
+import Foundation
+import SwiftUI
+
+class CompleteScreenViewModel: ObservableObject {
+    let session: SearchSession
+    @Published var selectedStream: Int?
+    @Published var isMapSelected: Bool = true
+    
+    init(session: SearchSession) {
+        self.session = session
+        selectedStream = session.streams.first?.id
+    }
+    
+    func stream(withID id: Int?) -> SearchSession.SearchSessionStream {
```

Comment:
2 things here:

1. ViewModels should not have this kind of in-out functions. We use bindings as outputs of viewmodels and functions as inputs. So viewmodel only has UI related input functions (that don't return anything) reflecting user interactions and `@Published` outputs which we bind into inside our views.

2. `SearchSessionStream` seems like a data layer object and should not be exposed in a viewmodel

Reply:
2. This function is used only to pass the right stream to chartView. How else should this be implemented?

Reply:
In this case @Binding is so comfortable to use, when we want to pass `selectedStream` as binding to `StaticSingleMeasurementView` so that when user taps the stream name in that view, they would see the right stream in chartView and the right stream name would have a border. I understand it could probably be done with a closure passed to this view, like `onStreamSelection`, but weren't bindings added to SwiftUI to make this process simpler?

Reply:
What are your thoughts on that? Could we discuss in on a call today?